src/pg_query_parse_plpgsql.c

```diff
@@ -354,13 +395,136 @@ PgQueryInternalPlpgsqlFuncAndError pg_query_raw_parse_plpgsql(CreateFunctionStmt
 	return result;
 }
 
+PgQueryInternalPlpgsqlFuncAndError pg_query_raw_parse_plpgsql_do(DoStmt* stmt)
```

Comment:
Since this function is almost 1:1 identical with `pg_query_raw_parse_plpgsql`, how about we instead detect the node type inside the existing `pg_query_raw_parse_plpgsql` function, like this:

```c
PgQueryInternalPlpgsqlFuncAndError pg_query_raw_parse_plpgsql(Node* stmt)
{
...
        PG_TRY();
	{
             if (IsA(stmt, CreateFunctionStmt)) {
                 result.func = compile_create_function_stmt((CreateFunctionStmt *) stmt);
             } else if (IsA(stmt, DoStmt)) {
                 result.func = compile_do_stmt((DoStmt *) stmt);
             } else {
                 elog(ERROR, "Unexpected node type for PL/pgSQL parsing: %d", nodeTag(stmt));
             }
...
}
```